bitnami/postgresql/templates/statefulset-slaves.yaml

```diff
@@ -83,9 +83,9 @@ spec:
             - -cx
             - |
               {{ if .Values.persistence.enabled }}
-              mkdir -p {{ .Values.persistence.mountPath }}/conf {{ .Values.persistence.mountPath }}/data
-              chmod 700 {{ .Values.persistence.mountPath }}/conf {{ .Values.persistence.mountPath }}/data
```

Comment:
When the configuration file is not mounted, it's autogenerated and writing permissions will be required. Suggestion:

- Create this macro in the `_helpers.tpl`:

```
{{/*
Return true if a configmap should be mounted with PostgreSQL configuration
*/}}
{{- define "postgresql.mountConfigmap" -}}
{{- if or (.Files.Glob "files/postgresql.conf") (.Files.Glob "files/pg_hba.conf") .Values.postgresqlConfiguration .Values.pgHbaConfiguration .Values.configurationConfigMap }}
    {- true -}}
{{- end -}}
{{- end -}}
```

- Use it here:

```
mkdir -p {{ .Values.persistence.mountPath }}/data {{ if not (include "postgresql.mountConfigmap" .) }} {{ .Values.persistence.mountPath }}/conf {{ end }}
chmod 700 {{ .Values.persistence.mountPath }}/data {{ if not (include "postgresql.mountConfigmap" .) }} {{ .Values.persistence.mountPath }}/conf {{ end }}
find {{ .Values.persistence.mountPath }} -mindepth 1 -maxdepth 1 {{ if not (include "postgresql.mountConfigmap" .) }} -not -name "conf" {{ end }} -not -name ".snapshot" -not -name "lost+found" | \
```

Reply:
I added it but had to change conditions for mkdir and chmod commands.

```yaml
  postgresqlConfiguration:
    listenAddresses: "'*'"
    port: "'5432'"
  postgresqlExtendedConf:
    maxConnections: 1024
  volumePermissions:
    enabled: true
    securityContext:
      runAsUser: "auto"
  securityContext:
    enabled: false
  shmVolume:
    chmod:
      enabled: false
```

I did some tests and it works fine. I think we are good to go :)